tiledb/CMakeLists.txt

```diff
@@ -317,6 +317,10 @@ if (TILEDB_AZURE)
       AzureSDK::AzureSDK
   )
   add_definitions(-DHAVE_AZURE)
+  
+  if(MSVC)
+    target_compile_definitions(TILEDB_CORE_OBJECTS PRIVATE -DCURL_STATICLIB=1 -Dazure_storage_lite_EXPORTS)
```

Comment:
Can we adjust this so the `CURL_STATICLIB` is only set when linking against the static library? We should be able to check this by checking the library ending like we do else where in the file. This will allow us to handle the static vs shared universally when curl is found on the system.